File: RxUI/src/main/java/com/vondear/rxui/view/wavesidebar/adapter/BaseViewHolder.java
package com.vondear.rxui.view.wavesidebar.adapter;

import android.support.v7.widget.RecyclerView;
import android.util.SparseArray;
import android.view.View;

public class BaseViewHolder extends RecyclerView.ViewHolder {

    /**
     * Views indexed with their IDs
     */
    private final SparseArray<View> mViews;


    public BaseViewHolder(View view) {
        super(view);
        this.mViews = new SparseArray<>();
    }

    @SuppressWarnings("unchecked")
    public  <V extends View> V findViewById(int viewId) {
        View view = mViews.get(viewId);
        if (view == null) {
            view = itemView.findViewById(viewId);
            mViews.put(viewId, view);
        }
        return (V) view;
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/wavesidebar/adapter/BaseViewHolder.kt
package com.tamsiree.rxui.view.wavesidebar.adapter

import android.util.SparseArray
import android.view.View
import androidx.recyclerview.widget.RecyclerView

open class BaseViewHolder(view: View?) : RecyclerView.ViewHolder(view!!) {
    /**
     * Views indexed with their IDs
     */
    private val mViews: SparseArray<View?> = SparseArray()
    fun <V : View?> findViewById(viewId: Int): V? {
        var view = mViews[viewId]
        if (view == null) {
            view = itemView.findViewById(viewId)
            mViews.put(viewId, view)
        }
        return view as V?
    }

}